app/uk/gov/hmrc/individualsbenefitsandcreditsapi/domains/integrationframework/IfWorkTaxCredit.scala

```diff
@@ -0,0 +1,52 @@
+/*
+ * Copyright 2020 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.individualsbenefitsandcreditsapi.domains.integrationframework
+
+import play.api.libs.functional.syntax._
+import play.api.libs.json.Reads.{max, min, verifying}
+import play.api.libs.json.{Format, JsPath, Reads}
+
+case class IfWorkTaxCredit(amount: Option[Double],
+                           entitlementYTD: Option[Double],
+                           paidYTD: Option[Double])
+
+object IfWorkTaxCredit {
+
+  val minPaymentValue = 0.0
```

Comment:
It would be easier to place these vars and some of the functions in a shared class and import. Would save us having to change the code in multiple places if a schema change alters this type?

Reply:
Added PatternsAndValidators trait and shared across IF domains